app/js/components/manageApps/InvalidZipUploadModal.jsx

```diff
@@ -0,0 +1,32 @@
+import React, { Component } from 'react';
+import {
+  Modal,
+  ModalHeader,
+  ModalTitle,
+  ModalClose,
+  ModalBody,
+  ModalFooter } from 'react-modal-bootstrap';
+
+const InvalidZipUploadModal = ({ isOpen, hideModal }) => {
+  return (
+    <Modal isOpen={isOpen} onRequestHide={hideModal}>
+      <ModalHeader>
+        <ModalClose onClick={hideModal}/>
+        <ModalTitle>Invalid Zip File</ModalTitle>
+      </ModalHeader>
+      <ModalBody>
+        <p>Could not successfully upload the zip file. The zip file should be a valid module</p>
```

Comment:
How do you tell that it is not a valid zip file?

Reply:
All valid `OWA` modules have a `manifest.webapp` file when unzipped. Targeting this specific content would quickly distinguish a valid zipped `OWA` module from an invalid one.

Reply:
So are you unzipping and checking for the manifest?

Reply:
Yes, that's what I'm doing.

Reply:
![screen shot 2017-10-13 at 12 42 17 pm 2](https://user-images.githubusercontent.com/22841496/31544736-2d33e458-b014-11e7-8303-091c81467775.png)

Reply:
Can you point me to where you are doing that from?